File: lib/models/restaurant.dart
import 'dart:convert';

List<Restaurants> restaurantsFromJson(String str) => List<Restaurants>.from(json.decode(str).map((x) => Restaurants.fromJson(x)));

String restaurantsToJson(List<Restaurants> data) => json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

class Restaurants {
  String? id;
  String? title;
  final String time;
  String? imageUrl;
  final String owner;
  final String code;
  final bool isAvailable;
  final bool pickup;
  final bool delivery;
  List<dynamic>? foods;
  final String? logoUrl;
  final double rating;
  final String ratingCount;
  final Coords coords;

  Restaurants({
    this.id,
    required this.title,
    required this.time,
    required this.imageUrl,
    required this.owner,
    required this.code,
    required this.isAvailable,
    required this.pickup,
    required this.delivery,
    this.foods,
    required this.logoUrl,
    required this.rating,
    required this.ratingCount,
    required this.coords,
  });

  factory Restaurants.fromJson(Map<String, dynamic> json) => Restaurants(
    id: json["_id"],
    title: json["title"],
    time: json["time"],
    imageUrl: json["imageUrl"],
    owner: json["owner"],
    code: json["code"],
    isAvailable: json["isAvailable"],
    pickup: json["pickup"],
    delivery: json["delivery"],
    foods: json["foods"] == null ? [] : List<dynamic>.from(json["foods"]!.map((x) => x)),
    logoUrl: json["logoUrl"],
    rating: json["rating"].toDouble(),
    ratingCount: json["ratingCount"],
    coords: Coords.fromJson(json["coords"]),
  );

  Map<String, dynamic> toJson() => {
    "_id": id,
    "title": title,
    "time": time,
    "imageUrl": imageUrl,
    "owner": owner,
    "code": code,
    "isAvailable": isAvailable,
    "pickup": pickup,
    "delivery": delivery,
    "foods": foods == null ? [] : List<dynamic>.from(foods!.map((x) => x)),
    "logoUrl": logoUrl,
    "rating": rating,
    "ratingCount": ratingCount,
    "coords": coords.toJson(),
  };
}

class Coords {
  final String id;
  final double latitude;
  final double longitude;
  final String address;
  final String title;
  final double latitudeDelta;
  final double longitudeDelta;

  Coords({
    required this.id,
    required this.latitude,
    required this.longitude,
    required this.address,
    required this.title,
    required this.latitudeDelta,
    required this.longitudeDelta,
  });

  factory Coords.fromJson(Map<String, dynamic> json) => Coords(
    id: json["id"],
    latitude: json["latitude"]?.toDouble(),
    longitude: json["longitude"]?.toDouble(),
    address: json["address"],
    title: json["title"],
    latitudeDelta: json["latitudeDelta"]?.toDouble(),
    longitudeDelta: json["longitudeDelta"]?.toDouble(),
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "latitude": latitude,
    "longitude": longitude,
    "address": address,
    "title": title,
    "latitudeDelta": latitudeDelta,
    "longitudeDelta": longitudeDelta,
  };
}

File: lab5-6backend-main/models/Restaurant.js
const mongoose = require('mongoose');
const generateOtp = require('../utils/otp_generator')
const restaurantSchema = new mongoose.Schema({
    title: {type: String , required: true},
    time: {type: String , required: true},
    imageUrl: {type: String , required: true},
    foods: {type: Array , default: []},
    pickup: {type: Boolean , default: true},
    delivery: {type: Boolean, default: true},
    owner: {type: String , required: true},
    isAvailable: {type: Boolean , default: false},
    code: {type: String , required: true},
    logoUrl: {type: String , required: true},
    rating: {
        type: Number,
        min: 1,
        max: 5,
        default: 3
    },
    ratingCount: {type: String, default: "12"},
    verification: {type: String ,default: "Pending", enum: ["Pending", "Verified", "Rejected"]},
    verificationMessage: {type: String, default: "Please allow up to 24 hours for your verification to be processed. You will receive a notification once your verification is complete."},
    coords: {
        id: {type: String, default: generateOtp() },
        latitude: {type: Number , required: true},
        longitude: {type: Number , required: true},
        latitudeDelta:{type: Number , default: 0.0122},
        longitudeDelta: {type: Number , default: 0.0221},
        address: {type: String , required: true},
        title: {type: String , required: true},
    },
    earnings: {type: Number, default: 0.868787}
});

const Restaurant = mongoose.model('Restaurant', restaurantSchema);

module.exports = Restaurant;
